### PR TITLE
XWIKI-16973: Spaces in file names in Zip files cause VFS Tree macro break

### PR DESCRIPTION
**Pull request by surli**
JIRA: 
  * https://jira.xwiki.org/browse/XWIKI-16793
  * https://jira.xwiki.org/browse/XWIKI-17244

This PR fixes two issues related to VFS macro:
  * the fact that we couldn't have a file with a space in a name in an archive (XWIKI-16973)
  * the fact that we couldn't rely on an archive with a space in its name (XWIKI-17244)

The provided fix consists mainly in not relying anymore on URI in VfsResourceReference, but on a new String reference attribute. For the problem of the space name in attachment name, it also needed some work to encode the name and decode it in VFS.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-vfs/xwiki-platform-vfs-api/src/main/java/org/xwiki/vfs/VfsResourceReference.java [hunk 1]
```diff
@@ -95,24 +105,77 @@ public VfsResourceReference(URI uri, String pathSegments)
...
     }
 
     /**
      * @return the URI to the VFS (e.g. {@code attach:space.page@file.zip}, {@code http://server/path/to/zip})
+     * @deprecated Since 12.3 this method shouldn't be used anymore, in favor of {@link #getReference}.
      */
+    @Deprecated
     public URI getURI()
     {
         return this.uri;
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-vfs/xwiki-platform-vfs-api/src/main/java/org/xwiki/vfs/VfsResourceReference.java:156]: Should this try to create a URI with the old logic when it's null for retro compatibility ?

Comment (surli) [reply xwiki-platform-core/xwiki-platform-vfs/xwiki-platform-vfs-api/src/main/java/org/xwiki/vfs/VfsResourceReference.java:156]: Could be safer indeed.

**Comment by surli** [comment]
@tmortagne change done for `#getURI`. I put the since on 11.10.5 too, since I was thinking about cherry-picking it for LTS too, let me know if you think it's too dangerous.

**Comment by tmortagne** [comment]
Given the number of changes you made it's definitely dangerous from VFS point of view. Now if there is enough tests to back it up...

**Comment by surli** [comment]
>  Now if there is enough tests to back it up...

Well I covered my changes, but I definitely might have missed some corner cases. Going to push and fix only for stable for now then.
